compiler/semantic/analyzer.go

```diff
@@ -81,9 +90,28 @@ func (a *analyzer) buildFrom(op dag.Op) (dag.Op, error) {
 	}
 	ops, err := a.semPool(pool)
 	if err != nil {
-		return nil, err
+		return err
+	}
+	s.Prepend(ops[0])
+	return nil
+}
+
+func StartsWithYield(s dag.Seq) bool {
```

Comment:
Nit: We call a dag.Seq parameter `seq` everywhere else.
```suggestion
func StartsWithYield(s dag.Seq) bool {
```